eessi/testsuite/tests/apps/QuantumESPRESSO.py

```diff
@@ -85,29 +64,14 @@ def set_increased_walltime(self):
         if self.ecut == max_ecut and self.nbnd == max_nbnd:
             self.time_limit = '60m'
 
-    @run_after('setup')
-    def run_after_setup(self):
-        """Hooks to run after the setup phase"""
-
-        # Calculate default requested resources based on the scale:
-        # 1 task per CPU for CPU-only tests, 1 task per GPU for GPU tests.
-        # Also support setting the resources on the cmd line.
-        if self.compute_device == DEVICE_TYPES[GPU]:
-            hooks.assign_tasks_per_compute_unit(test=self, compute_unit=COMPUTE_UNIT[GPU])
-        else:
-            hooks.assign_tasks_per_compute_unit(test=self, compute_unit=COMPUTE_UNIT[CPU])
-
-    @run_after('setup')
-    def request_mem(self):
-        memory_required = self.num_tasks_per_node * 0.9 + 4
-        hooks.req_memory_per_node(test=self, app_mem_req=memory_required * 1024)
-
-    @run_after('setup')
-    def set_omp_num_threads(self):
+    @run_after('init')
+    def set_compute_unit(self):
         """
-        Set number of OpenMP threads via OMP_NUM_THREADS.
-        Set default number of OpenMP threads equal to number of CPUs per task.
+        Set the compute unit to which tasks will be assigned:
+        one task per CPU core for CPU runs, and one task per GPU for GPU runs.
         """
-
-        self.env_vars['OMP_NUM_THREADS'] = self.num_cpus_per_task
-        log(f'env_vars set to {self.env_vars}')
+        device_to_compute_unit = {
```

Comment:
Maybe add a comment here what this does. I.e. make sure one task will be launched per CPU (core) for CPU-based runs, and one task is launched per GPU for GPU runs. For us it's clear, but someone new to the code might be a bit puzzled (it looks a bit like a `x = x` situation, but we know it's not).